src/org/opensolaris/opengrok/history/RCSRepository.java

```diff
@@ -127,7 +128,23 @@ void update() throws IOException {
     @Override
     boolean isRepositoryFor(File file) {
         File rcsDir = new File(file, "RCS");
-        return rcsDir.isDirectory();
+        if (!rcsDir.isDirectory()) {
+            return false;
+        }
+
+        // If there is at least one entry with the ',v' suffix,
+        // consider this a RCS repository.
+        String[] list = rcsDir.list(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                // Technically we should check whether the entry is a file
```

Comment:
I think the differences between list() and listFiles() are:
  - return type (String vs. File)
  - listFiles() does not  get "the directory itself and the directory's parent directory"

So, listFiles() will not help much and would likely incur higher overhead.